app/scripts/lib/snap-keyring/metrics.ts

```diff
@@ -0,0 +1,72 @@
+import { RestrictedControllerMessenger } from '@metamask/base-controller';
+import { KeyringControllerGetKeyringForAccountAction } from '@metamask/keyring-controller';
+import { GetSnap } from '@metamask/snaps-controllers';
+import { Snap } from '@metamask/snaps-utils';
+
+type AllowedActions = GetSnap | KeyringControllerGetKeyringForAccountAction;
+
+export type SnapAndHardwareMessenger = RestrictedControllerMessenger<
+  'SnapAndHardwareMessenger',
+  AllowedActions,
+  never,
+  AllowedActions['type'],
+  never
+>;
+
+export default async function getSnapAndHardwareInfoForMetrics(
+  getSelectedAddress: () => string,
+  getAccountType: (address: string) => Promise<string>,
+  getDeviceModel: (address: string) => Promise<string>,
+  messenger: SnapAndHardwareMessenger,
+) {
+  // If it's coming from a unit test, there's no messenger
+  // Will fix this in a future PR and add proper unit tests
+  if (!messenger) {
+    return {};
+  }
+
+  const selectedAddress = getSelectedAddress().toLowerCase();
+
+  const keyring: any = await getKeyringForAccount(selectedAddress);
+
+  const account = await getAccountFromAddress(selectedAddress);
+
+  const snap: Snap = (await messenger.call(
+    'SnapController:get',
+    account?.metadata.snap.id,
+  )) as Snap;
+
+  async function getAccountFromAddress(address: string) {
+    if (keyring.listAccounts) {
+      const accounts = await keyring.listAccounts();
+      return accounts.find(
+        (_account: any) => _account.address.toLowerCase() === address,
```

Comment:
we should compare the lowercase verson of both _account and address

Reply:
It's going to be lowercase, because it came from
`const selectedAddress = getSelectedAddress().toLowerCase();`
but yeah I moved the `toLowerCase()` to make it more explicit